File: mobile/examples/java/hello_world/Response.java
package io.envoyproxy.envoymobile.helloenvoy;

// Response is a class to handle HTTP responses.
public class Response {
  public final String title;
  public final String header;

  public Response(String title, String header) {
    this.title = title;
    this.header = header;
  }
}

File: mobile/examples/kotlin/shared/Response.kt
package io.envoyproxy.envoymobile.shared

// Response is a class to handle HTTP responses.
sealed class Response {
  fun fold(success: (Success) -> Unit, failure: (Failure) -> Unit) = when (this) {
    is Success -> success(this)
    is Failure -> failure(this)
  }
}

data class Success(val title: String, val header: String) : Response()

data class Failure(val message: String) : Response()



